test_util/integration_test.py

```diff
@@ -1340,3 +1349,148 @@ def wait_for_endpoint():
+            self._cluster._ssh_key_path,
+            self._cluster.slaves[0],
+            self.list_or_make(cmd)
+        )
+
+    @staticmethod
+    def list_or_make(v, split_fn=shlex.split):
+        return v if isinstance(v, list) else split_fn(v)
+
+    def _stop_mesos_agent(self):
+        self._run('systemctl stop dcos-mesos-slave')
+
+    def _start_mesos_agent(self, clear_state, restart=True):
+        if clear_state:
+            self._clear_agent_state()
+        self._run('systemctl start dcos-mesos-slave')
+
+    def _clear_agent_state(self):
+        # TODO: check this path
+        self._run('rm -rf /var/lib/mesos/slave')
+
+    def _run_volume_discovery(self, delete_only=False):
+        steps = (
+            'rm /var/lib/dcos/mesos-resources',
+            'systemctl start dcos-vol-discovery-priv-agent'
+        )
+        return self._multi_run(steps[:1] if delete_only else steps)
+
+    def _make_local_volume(self, size_mb, image_file):
+        return self._run('dd of={} if=/dev/null blocksize={}'.format(image_file, size_mb))
+
+    @contextlib.contextmanager
+    def _a_free_loop(self):
+        loop_device = self._run('losetup --find')
+        yield loop_device
+
+    def _attach_loop_back(self, mount_point, image_file):
+        with self._a_free_loop() as loop_device:
+            cmds = (
+                '/usr/bin/mkdir -p {}'.format(mount_point),
+                '/usr/sbin/losetup {} {}'.format(loop_device, image_file),
+                '/usr/sbin/mkfs -t ext4 {}'.format(loop_device),
+                '/usr/bin/mount {} {}'.format(loop_device, mount_point),
+            )
+            self._multi_run(cmds)
+
+    def _add_volumes_to_agent(self, vol_sizes):
+        for i, vol_size in enumerate(vol_sizes):
+            img = '/root/{}.img'.format(i)
+            mount_point = '/dcos/volume{}'.format(i)
+            self._make_local_volume(vol_size, img)
+            self._attach_loop_back(mount_point, img)
+            yield vol_size, img, mount_point
+
+    def _remove_volumes_from_agent(self, volumes):
+        for _, img, vol in volumes:
+            cmds = (
+                '/usr/bin/umount {}'.format(vol),
+                '/usr/sbin/losetup --detach {}'.format(vol),
+                'rm {}'.format(img),
+            )
+            self._multi_run(cmds)
+
+
+def get_state_json(cluster):
+    r = cluster.get('/mesos/master/slaves')
+    data = r.json()
+    slaves_ids = sorted(x['id'] for x in data['slaves'])
+
+    for slave_id in slaves_ids:
+        uri = '/slave/{}/slave%281%29/state.json'.format(slave_id)
+        r = cluster.get(uri)
+        data = r.json()
+        yield data
+
+
+def test_add_volume_noop(cluster):
+    agentm = AgentManipulator(cluster)
+    agentm._stop_mesos_agent()
+    added_volumes = agentm._add_volumes_to_agent((200, 200))
+    agentm._run_volume_discovery()
+    agentm._start_mesos_agent(clear_state=False)
+    agentm._remove_volumes_from_agent(added_volumes)
```

Comment:
I was a bit confused about why this line was here, until I realized you are doing some pre-assert cleanup work. So a comment might be nice... perhaps 'clean up mounted volumes before running test to ensure test does not mutate agent state'.